### ISSUE TITLE
Incorrect generic parameter kinds for generic parameter reordering suggestion

### ISSUE THREAD

**Issue by varkor** [rust-lang/rust#72816]
```rust
struct S<const X: (), T>;
```
produces:
```
error: type parameters must be declared prior to const parameters
 --> src/lib.rs:1:23
  |
1 | struct S<const X: (), T>;
  |         --------------^- help: reorder the parameters: lifetimes, then types: `<T, const X: ()>`
```
It should be: "types, then consts". This is due to:
https://github.com/rust-lang/rust/blob/a59264b01247836c70e24217e0d346b868387525/src/librustc_ast_passes/ast_validation.rs#L745-L753
which should not be gated on `const_generics`, and should take the particular parameters into account.

This issue has been assigned to @rakshith-ravi via [this comment](https://github.com/rust-lang/rust/issues/72816#issuecomment-636455269).

**Comment by teymour-aldridge** [comment]
@rakshith-ravi are you still interested in this (I'd like to take it if you aren't – no worries if you are, though)?

**Comment by rakshith-ravi** [comment]
Hey @teymour-aldridge. I'm already addressing this in my PR #72271.

**Comment by teymour-aldridge** [comment]
Oh – sorry! I didn't see it for some reason.

**Comment by rakshith-ravi** [comment]
No worries :)

**Comment by rakshith-ravi** [comment]
Fixed in #72271

**Comment by rakshith-ravi** [comment]
@varkor this can be closed
